extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -579,13 +578,15 @@ protected function getSearchConfigFromRequest() {
 
 		if ( $this->isCorporateWiki() ) {
 			$searchConfig->setLanguageCode( $request->getVal( 'resultsLang' ) );
-			if ( $searchConfig->getLanguageCode() !== 'en' ) {
-				$threshold = self::DEFAULT_NON_ENGLISH_WIKI_ARTICLE_THRESHOLD;
-				if ( in_array( 'staff', $this->wg->user->getEffectiveGroups() ) ) {
-					$threshold = $request->getVal( 'minArticleCount', self::DEFAULT_NON_ENGLISH_WIKI_ARTICLE_THRESHOLD );
-				}
-				$searchConfig->setXwikiArticleThreshold( $threshold );
+
+			$languageService = new \Wikia\Search\Language\LanguageService();
+			$languageService->setLanguageCode( $searchConfig->getLanguageCode() );
+			$wikiArticleThreshold = $languageService->getWikiArticlesThreshold();
+
+			if ( in_array( 'staff', $this->wg->user->getEffectiveGroups() ) ) {
+				$wikiArticleThreshold = $request->getVal( 'minArticleCount', $wikiArticleThreshold );
 			}
+			$searchConfig->setXwikiArticleThreshold( $wikiArticleThreshold );
```

Comment:
I'm not sure if we should set this if threshold is 0 (filter query), per:
https://github.com/Wikia/app/blob/DAT-2571/extensions/wikia/Search/classes/QueryService/Select/Dismax/InterWiki.php#L117